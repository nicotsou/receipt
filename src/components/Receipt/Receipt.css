.Receipt-list {
  list-style: none;
  padding: 0;
}

.Receipt-list-item {
  padding: 0;
  display: flex;
}

.Receipt-list-title {
  width: 70%;
}

.Receipt-list-price {
  text-align: right;
  width: 30%;
}

.Receipt-footer {
  text-align: center;
}