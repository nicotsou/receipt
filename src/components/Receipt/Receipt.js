import React from 'react';
import { bool, arrayOf, number, shape, string } from 'prop-types';
import map from 'lodash/map';

const renderLoading = () => {
  return <p>Loading Order...</p>;
}

const renderOrderList = (props) => {
  const items = map(props.items, item => {
    return (
      <li key={item.id}>
        <div>{item.quantity}x {item.name}</div>
        <div>{item.price}</div>
      </li>
    );
  });

  return (
    <ul>
      {items}
    </ul>
  );
}


const renderOrder = (props) => {
  return (
    <div>
      <p>Your order with #{props.orderId} contains the following items:</p>
      
      <ul>
        {renderOrderList(props)}
      </ul>

      <ul>
        <li>
          <span>Sub Total: </span>
          <span>{props.subTotalAmount}</span>
        </li>
        <li>
          <span>Tax Amount: </span>
          <span>{props.taxAmount}</span>
        </li>
        <li>
          <span>Total: </span>
          <span>{props.totalAmount}</span>
        </li>
      </ul>
    </div>
  );
}

export default function Receipt(props) {
  return (
    <section>
      <h1>Your Order</h1>
      {props.isLoading ? renderLoading(props) : renderOrder(props)}
    </section>
  );
}

Receipt.propTypes = {
  /**
   * We should display a loading indicator while the receipt is loading
   * @TODO Must go to an parent component, receipt should be clean
   */
  isLoading: bool,
  /**
   * A collection of order items to display
   */
  items: arrayOf(shape({
    id: string,
    name: string,
    price: number,
    quantity: number,
    discount: shape({
      amount: number,
      minItems: number
    }),
    subTotalAmount: number
  }))
};

Receipt.getDefaultProps = {
  isLoading: true
};