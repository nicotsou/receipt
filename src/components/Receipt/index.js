import React from 'react';

export default function Receipt(props) {
  return <div>Receipt here</div>;
}