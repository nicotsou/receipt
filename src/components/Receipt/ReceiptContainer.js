import { connect } from 'react-redux';

import {
  getReceiptItems,
  getTotal,
  getTax,
  getSubTotal,
  isAppLoading
} from '../../state/selectors';
import Receipt from './Receipt';

const mapStateToProps = state => ({
  isLoading: isAppLoading(state),
  orderId: state.receipt.orderId,
  items: getReceiptItems(state),
  subTotalAmount: getSubTotal(state),
  taxAmount: getTax(state),
  totalAmount: getTotal(state)
});

export default connect(
  mapStateToProps
)(Receipt);
