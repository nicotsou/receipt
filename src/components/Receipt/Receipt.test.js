import React from 'react';
import { shallow } from 'enzyme';

import Receipt from './Receipt';

it('should display a loading indicator by default', () => {
  const wrapper = shallow(
    <Receipt />
  );
  expect(wrapper).toMatchSnapshot();
});

it('should display a receipt', () => {
  const wrapper = shallow(
    <Receipt
      isLoading={false}
      items={[
        {
          id: 'apple',
          name: 'Apple',
          price: 0.50,
          quantity: 2
        },
        {
          id: 'orange',
          name: 'Orange',
          price: 0.90,
          quantity: 3
        },
        {
          id: 'bananna',
          name: 'Bananna',
          price: 0.60,
          quantity: 4
        },
        {
          id: 'papaya',
          name: 'Papaya',
          price: 2.5,
          quantity: 7,
          discount: {
            amount: 1,
            minItems: 3
          }
        }
      ]}
      orderId="#123"
      subTotalAmount={4.5}
      taxAmount={0.37}
      totalAmount={4.87}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
