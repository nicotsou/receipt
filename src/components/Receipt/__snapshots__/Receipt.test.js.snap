// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a loading indicator by default 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section>
    <h1>
        Your Order
    </h1>
    <div>
        <p>
            Your order with #
             contains the following items:
        </p>
        <ul
            className="Receipt-list"
        >
            
        </ul>
        <ul
            className="Receipt-list"
        >
            <li
                className="Receipt-list-item"
            >
                <span
                    className="Receipt-list-title"
                >
                    Sub Total: 
                </span>
                <span
                    className="Receipt-list-price"
                >
                    $ 
                </span>
            </li>
            <li
                className="Receipt-list-item"
            >
                <span
                    className="Receipt-list-title"
                >
                    Tax Amount: 
                </span>
                <span
                    className="Receipt-list-price"
                >
                    $ 
                </span>
            </li>
            <li
                className="Receipt-list-item"
            >
                <span
                    className="Receipt-list-title"
                >
                    Total: 
                </span>
                <span
                    className="Receipt-list-price"
                >
                    $ 
                </span>
            </li>
        </ul>
        <div
            className="Receipt-footer"
        >
            ^_^
        </div>
    </div>
</section>,
  "nodes": Array [
    <section>
      <h1>
            Your Order
      </h1>
      <div>
            <p>
                  Your order with #
                   contains the following items:
            </p>
            <ul
                  className="Receipt-list"
            >
                  
            </ul>
            <ul
                  className="Receipt-list"
            >
                  <li
                        className="Receipt-list-item"
                  >
                        <span
                              className="Receipt-list-title"
                        >
                              Sub Total: 
                        </span>
                        <span
                              className="Receipt-list-price"
                        >
                              $ 
                        </span>
                  </li>
                  <li
                        className="Receipt-list-item"
                  >
                        <span
                              className="Receipt-list-title"
                        >
                              Tax Amount: 
                        </span>
                        <span
                              className="Receipt-list-price"
                        >
                              $ 
                        </span>
                  </li>
                  <li
                        className="Receipt-list-item"
                  >
                        <span
                              className="Receipt-list-title"
                        >
                              Total: 
                        </span>
                        <span
                              className="Receipt-list-price"
                        >
                              $ 
                        </span>
                  </li>
            </ul>
            <div
                  className="Receipt-footer"
            >
                  ^_^
            </div>
      </div>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Receipt />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section>
          <h1>
                    Your Order
          </h1>
          <div>
                    <p>
                              Your order with #
                               contains the following items:
                    </p>
                    <ul
                              className="Receipt-list"
                    >
                              
                    </ul>
                    <ul
                              className="Receipt-list"
                    >
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Sub Total: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                        </span>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Tax Amount: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                        </span>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Total: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                        </span>
                              </li>
                    </ul>
                    <div
                              className="Receipt-footer"
                    >
                              ^_^
                    </div>
          </div>
</section>,
        "_debugID": 2,
        "_renderedOutput": <section>
          <h1>
                    Your Order
          </h1>
          <div>
                    <p>
                              Your order with #
                               contains the following items:
                    </p>
                    <ul
                              className="Receipt-list"
                    >
                              
                    </ul>
                    <ul
                              className="Receipt-list"
                    >
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Sub Total: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                        </span>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Tax Amount: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                        </span>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Total: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                        </span>
                              </li>
                    </ul>
                    <div
                              className="Receipt-footer"
                    >
                              ^_^
                    </div>
          </div>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Receipt />,
}
`;

exports[`should display a receipt 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section>
    <h1>
        Your Order
    </h1>
    <div>
        <p>
            Your order with #
            #123
             contains the following items:
        </p>
        <ul
            className="Receipt-list"
        >
            <li
                className="Receipt-list-item"
            >
                <div
                    className="Receipt-list-title"
                >
                    2
                    x 
                    Apple
                </div>
                <div
                    className="Receipt-list-price"
                >
                    $
                    0.5
                </div>
            </li>
            <li
                className="Receipt-list-item"
            >
                <div
                    className="Receipt-list-title"
                >
                    3
                    x 
                    Orange
                </div>
                <div
                    className="Receipt-list-price"
                >
                    $
                    0.9
                </div>
            </li>
            <li
                className="Receipt-list-item"
            >
                <div
                    className="Receipt-list-title"
                >
                    4
                    x 
                    Bananna
                </div>
                <div
                    className="Receipt-list-price"
                >
                    $
                    0.6
                </div>
            </li>
            <li
                className="Receipt-list-item"
            >
                <div
                    className="Receipt-list-title"
                >
                    7
                    x 
                    Papaya
                </div>
                <div
                    className="Receipt-list-price"
                >
                    $
                    2.5
                </div>
            </li>
        </ul>
        <ul
            className="Receipt-list"
        >
            <li
                className="Receipt-list-item"
            >
                <span
                    className="Receipt-list-title"
                >
                    Sub Total: 
                </span>
                <span
                    className="Receipt-list-price"
                >
                    $ 
                    4.5
                </span>
            </li>
            <li
                className="Receipt-list-item"
            >
                <span
                    className="Receipt-list-title"
                >
                    Tax Amount: 
                </span>
                <span
                    className="Receipt-list-price"
                >
                    $ 
                    0.37
                </span>
            </li>
            <li
                className="Receipt-list-item"
            >
                <span
                    className="Receipt-list-title"
                >
                    Total: 
                </span>
                <span
                    className="Receipt-list-price"
                >
                    $ 
                    4.87
                </span>
            </li>
        </ul>
        <div
            className="Receipt-footer"
        >
            ^_^
        </div>
    </div>
</section>,
  "nodes": Array [
    <section>
      <h1>
            Your Order
      </h1>
      <div>
            <p>
                  Your order with #
                  #123
                   contains the following items:
            </p>
            <ul
                  className="Receipt-list"
            >
                  <li
                        className="Receipt-list-item"
                  >
                        <div
                              className="Receipt-list-title"
                        >
                              2
                              x 
                              Apple
                        </div>
                        <div
                              className="Receipt-list-price"
                        >
                              $
                              0.5
                        </div>
                  </li>
                  <li
                        className="Receipt-list-item"
                  >
                        <div
                              className="Receipt-list-title"
                        >
                              3
                              x 
                              Orange
                        </div>
                        <div
                              className="Receipt-list-price"
                        >
                              $
                              0.9
                        </div>
                  </li>
                  <li
                        className="Receipt-list-item"
                  >
                        <div
                              className="Receipt-list-title"
                        >
                              4
                              x 
                              Bananna
                        </div>
                        <div
                              className="Receipt-list-price"
                        >
                              $
                              0.6
                        </div>
                  </li>
                  <li
                        className="Receipt-list-item"
                  >
                        <div
                              className="Receipt-list-title"
                        >
                              7
                              x 
                              Papaya
                        </div>
                        <div
                              className="Receipt-list-price"
                        >
                              $
                              2.5
                        </div>
                  </li>
            </ul>
            <ul
                  className="Receipt-list"
            >
                  <li
                        className="Receipt-list-item"
                  >
                        <span
                              className="Receipt-list-title"
                        >
                              Sub Total: 
                        </span>
                        <span
                              className="Receipt-list-price"
                        >
                              $ 
                              4.5
                        </span>
                  </li>
                  <li
                        className="Receipt-list-item"
                  >
                        <span
                              className="Receipt-list-title"
                        >
                              Tax Amount: 
                        </span>
                        <span
                              className="Receipt-list-price"
                        >
                              $ 
                              0.37
                        </span>
                  </li>
                  <li
                        className="Receipt-list-item"
                  >
                        <span
                              className="Receipt-list-title"
                        >
                              Total: 
                        </span>
                        <span
                              className="Receipt-list-price"
                        >
                              $ 
                              4.87
                        </span>
                  </li>
            </ul>
            <div
                  className="Receipt-footer"
            >
                  ^_^
            </div>
      </div>
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Receipt
        isLoading={false}
        items={
                Array [
                        Object {
                          "id": "apple",
                          "name": "Apple",
                          "price": 0.5,
                          "quantity": 2,
                        },
                        Object {
                          "id": "orange",
                          "name": "Orange",
                          "price": 0.9,
                          "quantity": 3,
                        },
                        Object {
                          "id": "bananna",
                          "name": "Bananna",
                          "price": 0.6,
                          "quantity": 4,
                        },
                        Object {
                          "discount": Object {
                            "amount": 1,
                            "minItems": 3,
                          },
                          "id": "papaya",
                          "name": "Papaya",
                          "price": 2.5,
                          "quantity": 7,
                        },
                      ]
        }
        orderId="#123"
        subTotalAmount={4.5}
        taxAmount={0.37}
        totalAmount={4.87}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "isLoading": false,
          "items": Array [
            Object {
              "id": "apple",
              "name": "Apple",
              "price": 0.5,
              "quantity": 2,
            },
            Object {
              "id": "orange",
              "name": "Orange",
              "price": 0.9,
              "quantity": 3,
            },
            Object {
              "id": "bananna",
              "name": "Bananna",
              "price": 0.6,
              "quantity": 4,
            },
            Object {
              "discount": Object {
                "amount": 1,
                "minItems": 3,
              },
              "id": "papaya",
              "name": "Papaya",
              "price": 2.5,
              "quantity": 7,
            },
          ],
          "orderId": "#123",
          "subTotalAmount": 4.5,
          "taxAmount": 0.37,
          "totalAmount": 4.87,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section>
          <h1>
                    Your Order
          </h1>
          <div>
                    <p>
                              Your order with #
                              #123
                               contains the following items:
                    </p>
                    <ul
                              className="Receipt-list"
                    >
                              <li
                                        className="Receipt-list-item"
                              >
                                        <div
                                                  className="Receipt-list-title"
                                        >
                                                  2
                                                  x 
                                                  Apple
                                        </div>
                                        <div
                                                  className="Receipt-list-price"
                                        >
                                                  $
                                                  0.5
                                        </div>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <div
                                                  className="Receipt-list-title"
                                        >
                                                  3
                                                  x 
                                                  Orange
                                        </div>
                                        <div
                                                  className="Receipt-list-price"
                                        >
                                                  $
                                                  0.9
                                        </div>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <div
                                                  className="Receipt-list-title"
                                        >
                                                  4
                                                  x 
                                                  Bananna
                                        </div>
                                        <div
                                                  className="Receipt-list-price"
                                        >
                                                  $
                                                  0.6
                                        </div>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <div
                                                  className="Receipt-list-title"
                                        >
                                                  7
                                                  x 
                                                  Papaya
                                        </div>
                                        <div
                                                  className="Receipt-list-price"
                                        >
                                                  $
                                                  2.5
                                        </div>
                              </li>
                    </ul>
                    <ul
                              className="Receipt-list"
                    >
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Sub Total: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                                  4.5
                                        </span>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Tax Amount: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                                  0.37
                                        </span>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Total: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                                  4.87
                                        </span>
                              </li>
                    </ul>
                    <div
                              className="Receipt-footer"
                    >
                              ^_^
                    </div>
          </div>
</section>,
        "_debugID": 4,
        "_renderedOutput": <section>
          <h1>
                    Your Order
          </h1>
          <div>
                    <p>
                              Your order with #
                              #123
                               contains the following items:
                    </p>
                    <ul
                              className="Receipt-list"
                    >
                              <li
                                        className="Receipt-list-item"
                              >
                                        <div
                                                  className="Receipt-list-title"
                                        >
                                                  2
                                                  x 
                                                  Apple
                                        </div>
                                        <div
                                                  className="Receipt-list-price"
                                        >
                                                  $
                                                  0.5
                                        </div>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <div
                                                  className="Receipt-list-title"
                                        >
                                                  3
                                                  x 
                                                  Orange
                                        </div>
                                        <div
                                                  className="Receipt-list-price"
                                        >
                                                  $
                                                  0.9
                                        </div>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <div
                                                  className="Receipt-list-title"
                                        >
                                                  4
                                                  x 
                                                  Bananna
                                        </div>
                                        <div
                                                  className="Receipt-list-price"
                                        >
                                                  $
                                                  0.6
                                        </div>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <div
                                                  className="Receipt-list-title"
                                        >
                                                  7
                                                  x 
                                                  Papaya
                                        </div>
                                        <div
                                                  className="Receipt-list-price"
                                        >
                                                  $
                                                  2.5
                                        </div>
                              </li>
                    </ul>
                    <ul
                              className="Receipt-list"
                    >
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Sub Total: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                                  4.5
                                        </span>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Tax Amount: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                                  0.37
                                        </span>
                              </li>
                              <li
                                        className="Receipt-list-item"
                              >
                                        <span
                                                  className="Receipt-list-title"
                                        >
                                                  Total: 
                                        </span>
                                        <span
                                                  className="Receipt-list-price"
                                        >
                                                  $ 
                                                  4.87
                                        </span>
                              </li>
                    </ul>
                    <div
                              className="Receipt-footer"
                    >
                              ^_^
                    </div>
          </div>
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Receipt
    isLoading={false}
    items={
        Array [
            Object {
              "id": "apple",
              "name": "Apple",
              "price": 0.5,
              "quantity": 2,
            },
            Object {
              "id": "orange",
              "name": "Orange",
              "price": 0.9,
              "quantity": 3,
            },
            Object {
              "id": "bananna",
              "name": "Bananna",
              "price": 0.6,
              "quantity": 4,
            },
            Object {
              "discount": Object {
                "amount": 1,
                "minItems": 3,
              },
              "id": "papaya",
              "name": "Papaya",
              "price": 2.5,
              "quantity": 7,
            },
          ]
    }
    orderId="#123"
    subTotalAmount={4.5}
    taxAmount={0.37}
    totalAmount={4.87}
/>,
}
`;
